hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -186,6 +187,16 @@ public String getString(String columnLabel) {
 
 	@Override
 	public boolean getBoolean(String columnLabel) {
+		final Object value = row.getValue( columnLabel );
+		wasNull = value == null;
+		if ( wasNull ) {
+			return false;
+		}
+
+		if( value instanceof BigInteger ) {
+			return ((BigInteger)value).intValue() > 0;
+		}
+
 		Boolean bool = row.getBoolean( columnLabel );
```

Comment:
No need to reread the row, this should work: 
```
Boolean bool = (Boolean) value;
```

Reply:
Personally, I would just do:
```
final Object value = row.getValue( columnLabel );
Boolean bool = value != null && value instanceof BigInteger
                               ? BigInteger.ONE.equals( (BigInteger) value) // TRUE only when 1
                                : (Boolean) value;
```

Reply:
@Barry, how the field is mapped in the database? It seems weird to return a `BigInteger` for a boolean.

Reply:
I mean, it seems like it would have to be a bug of some sort in the client....

Reply:
Yes, I think so, but it's hard to say without knowing which test fails without this change.

Reply:
To be honest, this seems a weird error for a client. It could also be that we are using the wrong type descriptor for a type in Oracle somewhere

Reply:
I haven't run any "get datatype" queries against Oracle yet... and yes it seems weird

Reply:
@blafond what's the test that's failing?

Reply:
I gave a quick look and the table is created with this query ([testSubclassViaAssociation](https://github.com/hibernate/hibernate-reactive/blob/284920c659e91d6ecb11b91320bb18a2c627682f/hibernate-reactive-core/src/test/java/org/hibernate/reactive/SingleTableInheritanceTest.java#L85)):

```
create table BookST (DTYPE varchar2(31 char) not null, id number(10,0) not null, published date, title varchar2(255 char), forbidden number(1,0), primary key (id))
```

The column `forbidden` is a primitive boolean and it's mapped as `number(1,0)`.
So I don't think the client is wrong returning a `BigInteger` but there is something not quite right with our types.

Reply:
Ah OK, thanks, now I see.

So now I recall that Oracle simply doesn't have a `boolean` type (which is nuts, but whatever).

So I guess this is simply another case where the Oracle JDBC driver would do this conversion automatically, but the Vert.x client doesn't.

Reply:
So @blafond, here's what I would do: I would try to call `getBoolean()` *first*, and if it throws, catch the exception and try calling `getInteger()` and converting the result. If that also fails, regrow the original exception.

Reply:
k. thx

Reply:
yep. that worked fine :)